PR 2 of 2: Visualizing CNS mutational signature exposures

sjspielman opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?

This is the second PR, with #1226, for updating the `mutational-signatures` module. 

#### What was your approach?

A notebook was created to make and export (currently in PDF) three plots, and hopefully two of those can be used for main text. 


#### What GitHub issue does your pull request address?

#1220 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?

+ Figures are in `plots/cns/`, and include:
  + Bubble plot
  + Bar plot of the literal exposures across broad histologies
  + Bar plot of mutations for the top-ten most mutated samples in the PBTA

Do we feel at least 2 of these could go to main text, or should further figure strategies be explored? Eg, codon mutations? 


#### Is there anything that you want to discuss further?

+ The gradient palette was used for the bubble plot, specifically considering only colors 3:8. The 9th is `na_color`, and the first two colors in the gradient palette are so light you can barely see them at all. Is this reasonable to only use 3:8?
+ The `ggsci` simpsons palette was used for the barplots. However, the default first 8 colors are not very distinguishable, so I used the first 6 + the last 2 colors in that palette. Is this reasonable? 
+ There are two ways to label the signatures, which can be seen here: https://signal.mutationalsignatures.com/explore/studyTissueType/1-6. I have labeled figures with the RegSig name, and _not_ `CNS_A`, etc. Do we prefer one naming scheme to another?
+ Note the overall project `README` still needs to be updated, and will be done within this PR before merge once the rest is agreed upon.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes


### Results

#### What types of results are included (e.g., table, figure)?

Three figures, both in the Rmd and exported to PDF. There is also a small table in the notebook giving some metadata about the top 10 mutated samples, which may be used in main text depending on which figures we incorporate.


#### What is your summary of the results?

N/A


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [ ] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jaclyn-taroni commented:
Before I review this, a general question: Is there a reason not to use some of the functions in `analyses/mutational-signatures/util/mut_sig_functions.R` like [`bubble_matrix_plot()`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/929a848deb11e0f5e8395d11c3def6a8aee637d9/analyses/mutational-signatures/util/mut_sig_functions.R#L104)  for example?

sjspielman commented:
The data was formatted much differently from how I had been approaching setting up the data. The bubble plot function consumes a specifically-structured data frame from another function in that utils file, which I did not use, so I ended up writing my own code inspired by some aspects of utils.

Of course if preferred, I definitely can integrate the existing functions by revisiting some of my initial data processing, which will necessarily also involve updating those functions.

jaclyn-taroni commented:
Going to summarize my takeaways from our Slack conversation earlier today: 

- We have a shared goal of minimizing duplication of functionality in this module
- The bar plots here require a different data format compared to the bar plots "earlier" in the module
- There's an option where we alter the earlier bar plots to better match the bar plots here, which probably includes updates to `util`
- To decide what to do, we start by identifying the visualization that will be included in the manuscript and to do that, you need to solicit feedback from multiple people!

With that being said, I'm probably going to wait to review this. If we didn't have the same takeaways, let me know!

sjspielman commented:
Going to convert this to a draft pending feedback solicitation!

sjspielman commented:
I have removed the barplot looking at exposures across short histology groups, and replaced with a new contender - a _boxplot_ of exposure distributions across relevant Fig 3 cancer groups. For this figure, exposures are visualized _for samples with non-zero exposures._ The percentage shown above each individual boxplot is the percent of samples in that cancer group with non-zero weights for that signature. For example, 43% of craniopharyngiomas have signature 11. 

Thoughts on this option? 

[exposures_boxplot.pdf](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/8144027/exposures_boxplot.pdf)

jaclyn-taroni commented:
@sjspielman - if 43% of craniopharyngiomas have non-zero values for signature 11, we'd expect then to plot the signature weight for n = (0.43 x craniopharyngioma samples), correct? If so, would we not benefit from showing the individual points given the low sample size?

jharenza commented:
Hi @sjspielman! Thanks so much for working on this 👏 . I have a few suggestions for the figures.

For the `bubble_plot.pdf`, would it be possible to make the medium and large circles much bigger to be able to see clearer differences in the proportions? I do have a bit of a hard time visually interpreting 0.6 vs 0.8.

>There are two ways to label the signatures, which can be seen here: https://signal.mutationalsignatures.com/explore/studyTissueType/1-6. I have labeled figures with the RegSig name, and not CNS_A, etc. Do we prefer one naming scheme to another?

I agree with the RefSig name here, but do you have any info on what MMR2 (vs MMR1 is) and what N2 vs N6 is? They were not a part of the original RefSig set and [this](https://signal.mutationalsignatures.com/explore/referenceCancerSignature/36) doesn't have an explanation.

I tend to like the stacked barplots within each cancer group for easier visual consumption, as in `samples_barplot.pdf`. Can we add a figure which plots these faceted by cancer group? It may be too busy, or go in the supplement, but may also point out some differences within cancer groups we can discuss.

In general, it may be helpful to order the axes by signature in numeric order 1 through N, then your MMR2 and N2, so they are in an easy-to-find order. I think I like the boxplots better than the bubble plot since the circles are hard to interpret as of now, so perhaps we do a group summary as the main takeaway (boxplot) and individual cancer summary. For the boxplot figure, can you add `(N = )` to the facet label?

sjspielman commented:
> For the bubble_plot.pdf, would it be possible to make the medium and large circles much bigger to be able to see clearer differences in the proportions? I do have a bit of a hard time visually interpreting 0.6 vs 0.8.

Yes, this is a tricky aspect. Re-sizing the bubbles is something that I played around a lot with, and the setting you see here is about the most differentiated I could make. We also probably want to update the code here more generally, since I am currently not using existing functions to make the bubble plot but would like to clean that up once we solidify an overall set of plots we'd like to use in the first place. 

> I agree with the RefSig name here, but do you have any info on what MMR2 (vs MMR1 is) and what N2 vs N6 is? They were not a part of the original RefSig set and [this](https://signal.mutationalsignatures.com/explore/referenceCancerSignature/36) doesn't have an explanation.

These are signatures that were identified as part of the original RefSig set, but they may not be directly referenced within the manuscript. However, they are in the supplementary tables [here](https://static-content.springer.com/esm/art%3A10.1038%2Fs43018-020-0027-5/MediaObjects/43018_2020_27_MOESM3_ESM.xlsx)

> I tend to like the stacked barplots within each cancer group for easier visual consumption, as in samples_barplot.pdf. Can we add a figure which plots these faceted by cancer group? It may be too busy, or go in the supplement, but may also point out some differences within cancer groups we can discuss.

Yes, I can work on something like this as another option for the paper.

> In general, it may be helpful to order the axes by signature in numeric order 1 through N, then your MMR2 and N2, so they are in an easy-to-find order. 

The current order is based on "other" signature names. These signatures have two names within the database - those I put int he figures, and alphabetical A-H. So they are currently arranged in A-H order. If this isn't particularly meaningful, then definitely we can move those tot he end.

>  I think I like the boxplots better than the bubble plot since the circles are hard to interpret as of now, so perhaps we do a group summary as the main takeaway (boxplot) and individual cancer summary. 

I tend to agree. 

> For the boxplot figure, can you add (N = ) to the facet label?
👍

sjspielman commented:
Replying to @jaclyn-taroni's comment..

> if 43% of craniopharyngiomas have non-zero values for signature 11, we'd expect then to plot the signature weight for n = (0.43 x craniopharyngioma samples), correct? If so, would we not benefit from showing the individual points given the low sample size?

First, are we cool that I removed those 0's in the first place? Removing the "rug of 0's" definitely improves the look of the viz but could be misleading. Hence, I added the fraction as possible middle ground, although with @jharenza comment about adding `N=` to facet labels, I could also change these fractions to actual numbers.

Second, are you thinking an overlayed sina/jitter over a boxplot (whose outliers are hidden)? I can also play around with a sina/jitter with an overlayed `stat_summary` (default mean+/-se pointrange geom seems reasonable here).

sjspielman commented:
To compare some different ideas, I updated the boxplot to instead be a sina plot with overlayed IQR box (no whiskers), show `N` in the facet labels, and show _all_ points instead of only the non-0. The x-axis order wasn't updated (yet?) but can be as needed. 
Any thoughts on this "overall" version? https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2117ec351a41df550ac14583bd33e4613e0c1613/analyses/mutational-signatures/plots/cns/exposures_boxplot.pdf

sjspielman commented:
I've made a few more (_draft concepts!! definitely not prime-time ready!!_ )plots to bring into discussion. Any and all thoughts on pursuing these strategies are welcome!

1) [A jitter plot](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/de8a93350ba28a61543ead18143940a404cf03f8/analyses/mutational-signatures/plots/cns/jitter_mutations_mb.pdf) of the mutation counts per mb across samples. 
2) [A stacked barplot](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/de8a93350ba28a61543ead18143940a404cf03f8/analyses/mutational-signatures/plots/cns/bad_counts_plot.pdf) of mutation counts per mb across cancer groups. I do not think this approach works well, so I tried the next approach...
3) [Two versions](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/de8a93350ba28a61543ead18143940a404cf03f8/analyses/mutational-signatures/plots/cns/barplot_concept_counts.pdf) of non-stacked plots of the _median_ mutation counts per mb across cancer groups, EITHER showing all samples or only exposed samples. These plots also have error bars for IQR, but they are quite tightly bounded so error bars may not be useful here.

jaclyn-taroni commented:
I like the sina plot with the IQR box best (https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2117ec351a41df550ac14583bd33e4613e0c1613/analyses/mutational-signatures/plots/cns/exposures_boxplot.pdf). Showing the 0s seems like _the move_ to me.

I do understand the impulse for the stacked barplot. I am a little worried that we're getting a bit too "far away" from the data, though.

sjspielman commented:
> I like the sina plot with the IQR box best (https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2117ec351a41df550ac14583bd33e4613e0c1613/analyses/mutational-signatures/plots/cns/exposures_boxplot.pdf). Showing the 0s seems like the move to me.

I tend to agree here, and with this display I like the zeros. My one question about it at this point is whether it's appropriately consistent with the overall plotting style for PBTA [as described](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/figures/README.md#overall-figure-theme) - "For 2+ group comparisons, we will use violin or boxplots with jitter." 

> I do understand the impulse for the stacked barplot. I am a little worried that we're getting a bit too "far away" from the data, though.

For this _particular_ data, I don't think stacked works well. But, we do want a figure in there showing mutation per mb or similar to accompany the sina/IQR figure about weights. The sina/IQR does a good job of ballparking proportion exposed, which the bubble plot does not do well (too hard to distinguish circle sizes, even under a variety of settings). We might consider the sina/IQR and a barplot of counts then for only exposed, aka the second one [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/de8a93350ba28a61543ead18143940a404cf03f8/analyses/mutational-signatures/plots/cns/barplot_concept_counts.pdf)?

jharenza commented:
> I like the sina plot with the IQR box best (https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/2117ec351a41df550ac14583bd33e4613e0c1613/analyses/mutational-signatures/plots/cns/exposures_boxplot.pdf). Showing the 0s seems like _the move_ to me.

I also agree with ☝️ 
 
> I do understand the impulse for the stacked barplot. I am a little worried that we're getting a bit too "far away" from the data, though.

For this, I was thinking the stacked barplots would be a per sample plot faceted by cancer group, rather than lumping, which I agree, does not help with telling the story. For example, S3A [in this paper](https://www.cell.com/cell-reports/pdfExtended/S2211-1247(19)31268-9). My main impetus for this was to see if there are any patterns within a cancer group, or perhaps mutually exclusive signatures which we can put into the supplement and describe. The sina plot doesn't give that level of granularity, but if we don't see anything interesting, that would be fine.

Another thought is to do a quick overall correlation plot of the signature weights to determine if in brain tumors we see any specific signatures correlating which either were not known to correlate or which are (eg- signatures 2 and 13, APOBEC, often correlate positively in the same samples - same paper, figure S5E - but we do not have those as part of the CNS signatures, thus if we see any correlating, it may be novel).

jharenza commented:
>The current order is based on "other" signature names. These signatures have two names within the database - those I put int he figures, and alphabetical A-H. So they are currently arranged in A-H order. If this isn't particularly meaningful, then definitely we can move those tot he end.

@sjspielman - yes, I think ordering by traditional numeric signatures (1, 3, 8, etc) would be more meaningful, and maybe a tilted axis text would help :)

sjspielman commented:
[Here's](https://github.com/sjspielman/OpenPBTA-analysis/blob/mutsig-07/analyses/mutational-signatures/plots/cns/exposures_per_sample_barplot.pdf) a first go of the sample barplot as discussed. Since this viz is very color-heavy, I used a colorblind scale instead of Simpsons. I also removed the specific specimen IDs since they are just noise in this figure. If there are some samples we want to highlight further for any reason, I would think that goes into a separate figure. 


Something I am wondering if this needs is some _ordering_ of the samples with facets, eg order them by a certain signature exposure?

jharenza commented:
> [Here's](https://github.com/sjspielman/OpenPBTA-analysis/blob/mutsig-07/analyses/mutational-signatures/plots/cns/exposures_per_sample_barplot.pdf) a first go of the sample barplot as discussed. Since this viz is very color-heavy, I used a colorblind scale instead of Simpsons. I also removed the specific specimen IDs since they are just noise in this figure. If there are some samples we want to highlight further for any reason, I would think that goes into a separate figure.
> 
> Something I am wondering if this needs is some _ordering_ of the samples with facets, eg order them by a certain signature exposure?

Wow, you're so quick! This looks really great and you can see some obvious differences across cancer groups, which make sense (MMR in DMG/HGG, including only MMR in a few samples - we should double check those are the hypermutated samples!). I wonder if we should just order descending by signature 1, which is a universal signature - the more of signature 1 and the less of others, the closer it is to a "normal" sample. Maybe by MMR or sig 3 (BRCA) would be more interesting since I see that while MMR in HGG/DMG is more universal, it not in all samples in other groups - this possibly could associate with subtype or phase of therapy (eg progression/relapse having more mutations and possibly more MMR?).

sjspielman commented:
Sounds great, I'll look into 1/3/MMR options for sorting and check about hypermutated samples in this notebook.

sjspielman commented:
I've updated the samples barplot to order by signature 1. Among all the options, this one looked the best in part because signature 1 occurs more frequently overall.

I also explored some aspects of how signatures and mutation burdens may relate to tumor descriptors and subtypes. I didn't saved any of those figures specifically because nothing jumped out at me very strongly, but you can see [in the previewed notebook](http://htmlpreview.github.io/?https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/d9a470f08e6f06bbf11c5b7f4cf4452eecf79cef/analyses/mutational-signatures/07-plot_cns_fit.nb.html) with full context if we think there's more to potentially explore here.

jharenza commented:
> I've updated the samples barplot to order by signature 1. Among all the options, this one looked the best in part because signature 1 occurs more frequently overall.

Thanks! Yes, this makes sense and looks good. It is interesting to me that some samples have other signatures which are so strong they then attenuate Signature 1.

> I also explored some aspects of how signatures and mutation burdens may relate to tumor descriptors and subtypes. I didn't saved any of those figures specifically because nothing jumped out at me very strongly, but you can see [in the previewed notebook](http://htmlpreview.github.io/?https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/d9a470f08e6f06bbf11c5b7f4cf4452eecf79cef/analyses/mutational-signatures/07-plot_cns_fit.nb.html) with full context if we think there's more to potentially explore here.

What I was thinking about when I mentioned this was if we look specifically within a cancer group from your plot which are now arranged by Sig 1, as samples lose Sig 1, are those more progressive/relapse/post mortem or a specific subtype? I would hypothesize those which lose Sig 1 and gain others might be more enriched for progressive/relapse/post mortem because of possibly more mutations occurring in those samples. Not sure how much we want to explore here, though. 

>including only MMR in a few samples - we should double check those are the hypermutated samples!

For this, can we do a simple check that all samples with MMR2 signature weight == 1.0 have a high (hyper- or ultrahyper-mutant TMB)?

sjspielman commented:
> What I was thinking about when I mentioned this was if we look specifically within a cancer group from your plot which are now arranged by Sig 1, as samples lose Sig 1, are those more progressive/relapse/post mortem or a specific subtype (I might only explore MB subtypes and/or HGAT H3 mutant vs not mutant here for simplicity)? I would hypothesize those which lose Sig 1 and gain others might be more enriched for progressive/relapse/post mortem because of possibly more mutations occurring in those samples. Not sure how much we want to explore here, though.

Ah, ok. So this is effectively asking whether the _proportion of signature 1 exposure_ (of all exposures) for a given sample is related to subtypes etc, yes?

>For this, can we do a simple check that all samples with MMR2 signature weight == 1.0 have a high (hyper- or ultrahyper-mutant TMB)?

Can do!

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
````diff
@@ -0,0 +1,364 @@
+```{r}
+highly_mutated <- df_signature_mutations_per_mb %>%
+  dplyr::arrange(-mutations_per_mb_per_signature) %>%
+  dplyr::filter(mutations_per_mb_per_signature > 1) %>%
````
Do you have a reference for this threshold?

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,364 @@
+  dplyr::mutate(Kids_First_Biospecimen_ID = factor(Kids_First_Biospecimen_ID, levels = samples_in_order)) %>%
+  ggplot2::ggplot() +
+  ggplot2::aes(x = Kids_First_Biospecimen_ID, 
+               y = exposure, 
```
So would we expect the exposures to _mostly_ sum to 1 and when that is not the case, it is due to filtering out low confidence assignments (whatever the `deconstructSigs` equivalent is for that concept)?

sjspielman replied:
Not necessarily. There is potential for unknown weight across samples, which is also embedded in the RDS and can be explored. The function does not require that every mutation be assigned to a signature, but rather to determine the extent to which each signature is present. A quick look at this shows this distribution for unknown proportions across samples - 
```
 Min.   :0.000000  
 1st Qu.:0.005315  
 Median :0.042708  
 Mean   :0.047542  
 3rd Qu.:0.072105  
 Max.   :0.255552 
```

sjspielman replied:
Actually, something we might want to do is add "unknown" as a signature category to be explicit about this in the viz. What do you think?

jaclyn-taroni replied:
That makes sense to me!

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
````diff
@@ -0,0 +1,364 @@
+  
+```
+
+Which samples specifically are those with extremely high mutations in these signatures? Mostly HGG, which are all H3 wildtype without a particular tp53 pattern.
````
It looks like this PR was marked ready for review _before_ this comment came in.

jaclyn-taroni replied:
And now I see that you planned to address this yesterday... oops, sorry!

sjspielman replied:
Yes, this is being taken out! Got a bit pushed back with training :)

jaclyn-taroni reviewed (commented):
I reviewed this because I didn't realize that there were remaining comments to be addressed and for that I apologize! I expect many of the comments to still be relevant. I also think there might be some "vestigial code" in here (or I missed something) because I'm not sure we use the mutations per Mb info anywhere?

sjspielman commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,364 @@
+
+# Read in exposures from CNS signature fitting and tibble it up
+raw_exposures <- readr::read_rds(fitted_file)
+decon_mean <- lapply(raw_exposures, "[[", "weights")
```
This is per sample. It's the expectation from decomposition, as opposed to lower/upper CI bounds. Nothing is being averaged across samples, but this variable name was left over from when I was using `sigfit` so it's confusing for the context.

sjspielman commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,364 @@
+}
+
+# Input files
+fitted_file    <- file.path(analysis_path, "results", "fitted_exposures_signal-cns_deconstructSigs.rds")
```
Fixed.

sjspielman commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
````diff
@@ -0,0 +1,364 @@
+The barplot below shows the proportion of samples with non-zero exposures to each signature. This is effectively the number of non-zero points in the sina/IQR plot.
+
+```{r proportion, fig.width = 12, fig.height = 5}
+df_signature_mutations_per_mb %>%
````
Yeah, this was a quick exploratory figure that I don't see going anywhere. Can't hurt to clean up a little bit anyways!

sjspielman commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,297 @@
+samples_barplot_file <- file.path(analysis_path, "plots", "cns", "samples_barplot.pdf")
+
+# The real signature names: https://signal.mutationalsignatures.com/explore/studyTissueType/1-6
+signal_names <- c("11", "1", "N6", "8", "MMR2", "18", "19", "3") # In order A, B, C...
```
Yes it will change my approach, but your suggestion is better I think and a bit safer for swapping out labeling.

sjspielman commented:
The notebook has now been massively cleaned up and is ready for another look @jaclyn-taroni !
+ Incorporate review feedback, including removing vestigial code 
+ Include an "other" category in viz representing exposures UNASSIGNED to the 8 CNS
+ Finalize and export barplot of proportion of exposed signatures
+ Exploratory analyses without exported viz/tables because no strong trends were observed:
    + Explore signature 1 presence/absence and potential correlates
    + Explore MMR2-dominant samples and potential correlates

Note, the README will be finalized with results once we agree on them.

jharenza commented:
@sjspielman can you make your barplot and sina plots 5 rows x 2 columns? This will fit more perfectly in Figure 3.

sjspielman commented:
Regarding `ggforce`, this package is added to the [Dockerfile _in my branch_](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/bb7e72868b11deccc707044d9a2e6e338c4536ce/Dockerfile), so for now you can only run the code if you checkout my branch. Once this gets merged into master, `ggforce` will be in the docker image more generally.

I'll add to the notebook also:
+ Plots in a more vertical layout
+ Info about those 4 highly mutated samples
+ Expand the signature 1 plots to look at all cancer groups. For now I'll keep these in the notebook (not exported images)

sjspielman commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
````diff
@@ -0,0 +1,364 @@
+
+
+```{r boxplot, fig.width = 11, fig.height = 4.5}
+cancer_groups_to_plot <- c(
+  "Diffuse midline glioma",
+  "Low-grade glioma astrocytoma",
+  "Craniopharyngioma",
+  "High-grade glioma astrocytoma",
+  "Ganglioglioma",
+  "Medulloblastoma",
+  "Meningioma",
+  "Ependymoma",
+  "Schwannoma",
+  "Dysembryoplastic neuroepithelial tumor"
+)
````
Linking this script where groups were defined https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/interaction-plots/scripts/03-plot_interactions.R.

jaclyn-taroni commented:
@sjspielman can you push a version of the notebook with the output rendered please? I don't think all the plots are saved as PDFs and I'd like to take a look at them as I review 👀

sjspielman commented:
I've made some of the changes, except I can't get the samples barplot vertical. It becomes a very dizzying "optical illusion" with all the tight bars for high N samples if the plot gets narrower than this.

sjspielman commented:
> can you push a version of the notebook with the output rendered please? I don't think all the plots are saved as PDFs and I'd like to take a look at them as I review 

The most recent push should have everything rendered in Rmd. I haven't exported the exploratory plots, but can quickly do those exports now, stay tuned for another commit in a few min.

jaclyn-taroni commented:
@jharenza the discussion here is getting long (there are a bunch of hidden items!) which is usually a sign that we should get this in and then file issues with what else we'd like to see.

sjspielman commented:
Noting CI failed because of the CI dataset. I will add a param to this file to only run that particular plot if not CI.

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,485 @@
+
+### Barplot of sample exposures faceted across cancer groups
+
+This plot shows the literal exposures for each samples.
```
I find this wording confusing.

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,485 @@
+               y = exposure, 
+               fill = signature) + 
+  ggplot2::geom_col(color = "black", size = 0.05) + 
+  ggplot2::labs(x = "Sample", y = "Signature exposure", fill = "RefSig Signature") + 
```
This y-axis label appears to be using "`sigfit` parlance" – I wonder if that's potentially confusing to an audience that's familiar with `deconstructSigs`? I might be overthinking it.

sjspielman replied:
Do you mean the term "exposure," vs something like "weights"?

sjspielman replied:
Noting I ended up changing labels with "exposure" to use the word "weights"

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,485 @@
+Which samples have MMR ~ 1, and are these hypermutators? 
+This analysis is discussed [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1227#issuecomment-1064493652).
+
+Note that this visualiation is strictly exploratory and not exported. 
```
```suggestion
Note that this visualization is strictly exploratory and not exported. 
```

jaclyn-taroni commented on analyses/mutational-signatures/07-plot_cns_fit.Rmd:
```diff
@@ -0,0 +1,485 @@
+  dplyr::inner_join(subtypes) %>%
+  dplyr::group_by(Kids_First_Biospecimen_ID) %>%
+  tidyr::spread(signature, exposure) %>%
+  knitr::kable()
```
I do not find this table to be particularly easy to read. I think it might be how the signature numbers are aligned and the inconsistent width. 

This is a later problem _but_ it's kind of difficult to tell if these samples show a consistent pattern that's _any different_ from all other tumors in that subtype. I could imagine an exposures stacked bar plot (within HGG, H3 wildtype) that's faceted by hypermutated vs. not.

jaclyn-taroni replied:
If/when we do this, I'd maybe recommend splitting everything in the explore section into its own notebook just to keep things more digestible. You'd presumably want to save some things from earlier in the notebook in tabular form.

sjspielman replied:
Noting I removed the pipe into `knitr::kable()`.

jaclyn-taroni reviewed (approved):
Let's get this in! I have some _future thoughts_ ™️  that I've added, but let's get this in once https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1227/files#r827964059 is addressed.
